tools/hardware/config.py

```diff
@@ -3,7 +3,8 @@
 #
 # SPDX-License-Identifier: GPL-2.0-only
 #
-from typing import List, Set
+
+from __future__ import annotations
```

Comment:
We need this, so we can write `def foo() -> SomeType: ...` instead of having to use quotes `def foo() -> 'SomeType': ...`. This was supposed to become possible by default in 3.10, but has been postponed.

Reply:
We have Python 3.8 in the container now, so after some more testing it turn out this is no longer necessary in general. There are just some corner cases where I' trying to figure out where this is needed. Having this does not harm, but it allows using some feature newer python version will make the default. So while this is nice, there is also a certain danger to break the oldest version we support . Ideally, we'd have a `trustworthysystems/sel4:oldest` container that is used in a CI jobs also  to test thing are still fine in the oldest configuration we support.
Unless we don't have this, it might actually be better not to have this line anywhere unless it is absolutely required.